core-common/src/main/java/org/mapstruct/MappingInheritanceStrategy.java

```diff
@@ -0,0 +1,48 @@
+/**
+ *  Copyright 2012-2015 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct;
+
+/**
+ * The strategy to use for applying {@code @Mapping} configurations of prototype methods in the interface specified with
+ * {@link Mapper#config()}.
+ *
+ * @author Andreas Gudian
+ */
+public enum MappingInheritanceStrategy {
+    /**
+     * Apply the {@code @Mapping} configuration only if the prototype method is explicitly referenced using
+     * {@link InheritConfiguration}.
+     */
+    WHEN_EXPLICITLY_REFERENCED,
+
+    /**
+     * Apply the {@code @Mapping} configuration if source and target types of the prototype method are assignable from
+     * the types of a given mapping method.
+     */
+    AUTOMATICALLY_WHEN_ALL_TYPES_MATCH,
```

Comment:
@agudian, what is intended to happen when using this strategy and several prototype methods are matching? On first thought I'd say that method should be marked as erroneous, requiring to explicitly specify a prototype method via `@InheritConfiguration. WDYT?